scipy/stats/stats.py

```diff
@@ -611,27 +611,30 @@ def mode(a, axis=0):
                       [4, 7, 5, 9]])
     >>> from scipy import stats
     >>> stats.mode(a)
-    (array([[ 3.,  1.,  0.,  0.]]), array([[ 1.,  1.,  1.,  1.]]))
+    (array([[3, 1, 0, 0]]), array([[1, 1, 1, 1]]))
 
     To get mode of whole array, specify axis=None:
 
     >>> stats.mode(a, axis=None)
-    (array([ 3.]), array([ 3.]))
+    (array([3]), array([3]))
 
     """
     a, axis = _chk_asarray(a, axis)
-    scores = np.unique(np.ravel(a))       # get ALL unique values
-    testshape = list(a.shape)
-    testshape[axis] = 1
-    oldmostfreq = np.zeros(testshape, dtype=a.dtype)
-    oldcounts = np.zeros(testshape)
-    for score in scores:
-        template = (a == score)
-        counts = np.expand_dims(np.sum(template, axis), axis)
-        mostfrequent = np.where(counts > oldcounts, score, oldmostfreq)
-        oldcounts = np.maximum(counts, oldcounts)
-        oldmostfreq = mostfrequent
-    return mostfrequent, oldcounts
+    if a.size == 0:
+        return np.array([]), np.array([])
+    else:
```

Comment:
it's better to remove this ``else`` and view the ``if`` clause as an isolated check. This is also what's done for such checks in other functions. If you do it like this and have multiple checks on input parameters, the indentation of the code below keeps increasing.